### PR TITLE
Gate docs build

### PR DESCRIPTION
**Pull request by danewalton**

### CODE REVIEW

File: doxygen/style.css [hunk 1]
```diff
@@ -0,0 +1,1796 @@
...
+}
+
+table.DocNodeLTR {
+   width: auto;
+   margin-right: auto;
+   margin-left: 0;
+}
+
+tt, code, kbd, samp
+{
```
Comment (hihigupt) [inline doxygen/style.css:1788]: is there format of this file ? 
like always start { from next line?

**Comment by hihigupt** [comment]
Is there some link that I should be verifying if docs are looking good ?

**Comment by danewalton** [comment]
> Is there some link that I should be verifying if docs are looking good ?

https://azure.github.io/azure-iot-middleware-freertos/

File: doxygen/style.css (hunk 1 above)
Comment (danewalton) [reply doxygen/style.css:1788]: I took this straight from the embedded so I'm not entirely sure.

File: .github/workflows/ci_tests_linux.yml [hunk 2]
```diff
@@ -23,3 +23,21 @@ jobs:
...
                          .github/scripts/ci_tests.sh"
+
+  push-docs:
+
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install and Build
```
Comment (ericwol-msft) [inline .github/workflows/ci_tests_linux.yml:34]: better name?

File: doxygen/Doxyfile [hunk 3]
```diff
@@ -0,0 +1,2454 @@
+# Doxyfile 1.8.13
```
Comment (ericwol-msft) [inline doxygen/Doxyfile:1]: do you really want doxygen  a root dir?

File: source/include/azure_iot.h [hunk 4]
```diff
@@ -57,11 +60,19 @@ typedef struct AzureIoTMessageProperties
     struct
     {
         az_iot_message_properties xProperties;
-    } _internal;
+    } _internal; /**< @brief Internal to the SDK */
 } AzureIoTMessageProperties_t;
 
+/**
+ * @brief The time function to be used by the SDK for MQTT connections.
```
Comment (ericwol-msft) [inline source/include/azure_iot.h:67]:
```suggestion
 * @brief The platform get time function to be used by the SDK for MQTT connections.
```

File: source/include/azure_iot.h [hunk 5]
```diff
@@ -57,11 +60,19 @@ typedef struct AzureIoTMessageProperties
...
+ *
+ * @note Must return the time since Unix epoch.
+ */
 typedef uint64_t ( * AzureIoTGetCurrentTimeFunc_t )( void );
 
+/**
+ * @brief The HMAC function used by the SDK to generate SAS keys.
```
Comment (ericwol-msft) [inline source/include/azure_iot.h:74]:
```suggestion
 * @brief The HMAC256 function used by the SDK to generate SAS keys.
```

File: source/include/azure_iot_hub_client.h [hunk 6]
```diff
@@ -23,17 +23,26 @@
...
+ * @brief MQTT quality of service values used for messages.
+ */
 typedef enum AzureIoTHubMessageQoS
 {
     eAzureIoTHubMessageQoS0 = 0, /** Delivery at most once. */
     eAzureIoTHubMessageQoS1 = 1  /** Delivery at least once. */
 } AzureIoTHubMessageQoS_t;
 
+/**
+ * @brief Enumeration to dictate message types.
```
Comment (ericwol-msft) [inline source/include/azure_iot_hub_client.h:44]:
```suggestion
 * @brief Enumeration to dictate Azure IoT message types.
```

File: source/include/azure_iot_hub_client.h [hunk 7]
```diff
@@ -197,6 +216,10 @@ typedef struct AzureIoTHubClientOptions
                                                         *   Can be NULL if user does not want to be notified.*/
 } AzureIoTHubClientOptions_t;
 
+/**
+ * @struct AzureIoTHubClient_t
+ * @brief Azure IoT Hub Client used to manage connections and features for IoT Hub.
```
Comment (ericwol-msft) [inline source/include/azure_iot_hub_client.h:221]:
```suggestion
 * @brief Azure IoT Hub Client used to manage connections and features for Azure IoT Hub.
```

File: source/include/azure_iot_hub_client.h [hunk 8]
```diff
@@ -59,6 +71,9 @@ typedef enum AzureIoTHubClientResult
     eAzureIoTHubClientFailed,             /**< There was a failure. */
 } AzureIoTHubClientResult_t;
 
+/**
+ * @brief Status codes for IoT Hub responses.
```
Comment (ericwol-msft) [inline source/include/azure_iot_hub_client.h:75]:
```suggestion
 * @brief Status codes for Azure IoT Hub responses.
```

File: source/interface/azure_iot_mqtt.h [hunk 9]
```diff
@@ -20,30 +20,37 @@
...
+ */
 typedef enum AzureIoTMQTTQoS
 {
     eAzureIoTMQTTQoS0 = 0, /** Delivery at most once. */
     eAzureIoTMQTTQoS1 = 1, /** Delivery at least once. */
     eAzureIoTMQTTQoS2 = 2  /** Delivery exactly once. */
 } AzureIoTMQTTQoS_t;
 
+/**
+ * @brief Suback ack status states.
```
Comment (ericwol-msft) [inline source/interface/azure_iot_mqtt.h:52]:
```suggestion
 * @brief MQTT Suback ack status states.
```

**Review by ericwol-msft — APPROVED** [review]

**Review by hihigupt — APPROVED** [review]

**Pull request merged**